stress_test_model_loan_book.R

```diff
@@ -547,10 +560,29 @@ for (i in seq(1, nrow(transition_scenarios))) {
       year == start_year,
       technology %in% technologies,
       scenario_geography == scenario_geography_filter
+    ) #%>%
+    # distinct(investor_name, portfolio_name, company_name, ald_sector, technology,
+    #          scenario_geography, year, plan_carsten, plan_sec_carsten, term,
+    #          PD_0)
+
+  financial_data_loans_pd <- financial_data_loans %>%
+    select(company_name, company_id, ald_sector, technology, pd) %>%
+    distinct(across(everything()))
+
+  plan_carsten_loanbook <- plan_carsten_loanbook %>%
+    # ADO 879: removed company id from join, but should be re-introduced later on
+    left_join(financial_data_loans_pd, by = c("company_name", "ald_sector", "technology"))
+  # TODO: what to do with entries that have NAs for pd?
+  # TODO: kick out NAs and record the diff
+  loanbook_annual_profits <- loanbook_annual_profits %>%
```

Comment:
actually, since i noticed this for for equity_pacta_result also,  when is company id na while name is there? Also, would that not created problems in earlier steps, e.g. in any grouping operation that uses company_id?

Reply:
This says NAs for PD not ID.. (which essentially comes back to the left join question)

on the id question: there should not be any companies with a name but no id. I think I may have seen such a case in some raw data file (probably financial data) at some point, but they probably do not get carried through to this stage of the code. I realise this is all pretty vague, but in the end this would best be handled at the import

Reply:
ok, resolving because not relevant here.